src/handlers/query-user.ts

```diff
@@ -0,0 +1,62 @@
+import { Context } from "../types/context";
+
+export async function queryUser(context: Context, username: string) {
+  const {
+    octokit,
+    payload,
+    adapters: { supabase },
+    config,
+  } = context;
+  const body: string[] = [];
+  try {
+    const {
+      data: { id },
+    } = await octokit.users.getByUsername({
+      username,
+    });
+    if (!config.allowPublicQuery && payload.organization) {
+      const { status } = await octokit.orgs.checkMembershipForUser({
+        username,
+        org: payload.organization.login,
+      });
+      // @ts-expect-error Somehow typing seems wrong but according to
+      // https://docs.github.com/en/rest/orgs/members?apiVersion=2022-11-28#check-organization-membership-for-a-user--status-codes
+      // 204 means the user is part of the Organization
+      if (status !== 204) {
+        body.push(`\`\`\`User ${username} cannot request another user as it is not member of the organization.\`\`\``);
+        await octokit.issues.createComment({
+          body: body.join("\n"),
+          owner: payload.repository.owner.login,
+          repo: payload.repository.name,
+          issue_number: payload.issue.number,
+        });
+        return;
+      }
+    }
+    const access = await supabase.access.getAccess(id);
+    const wallet = await supabase.wallet.getWallet(id);
+    if (!access && !wallet) {
+      body.push(`\`\`\`
+User information for ${username} was not found.
+\`\`\``);
+    } else {
+      body.push(`
+| Property | Value |
+|----------|-------|`);
+      if (wallet) {
+        body.push(`| Wallet | ${wallet.address} |`);
+      }
+      if (access) {
+        body.push(`| Access | ${access.multiplier_reason} |`);
+      }
+    }
+    await octokit.issues.createComment({
+      body: body.join("\n"),
+      owner: payload.repository.owner.login,
+      repo: payload.repository.name,
+      issue_number: payload.issue.number,
+    });
+  } catch (e) {
+    context.logger.error("Could not query user.", e);
```

Comment:
The logger should automatically handle posting the comment as well I believe. That's how I designed my logger in the old code.

Reply:
But does it make sense to post comment for every log? We should make it like the old bot where the logger returned an error and if it's thrown then a comment will be posted

Reply:
Thought about this, might be redundant with https://github.com/ubiquibot/plugins-wishlist/issues/12
I think since the kernel is the only one aware of the state of each plugin, should be responsible for posting **critical** errors. Query a non-existing user doesn't seems critical, so just posting the comment with the result seems okay?

And agree that in the end there should be only one log with all the errors together. Or we could throw on critical level as well.

Reply:
> so just posting the comment with the result seems okay?

There's an environment variable in the version one of the bots that sets what log level should be commented. For example, `info` logs don't necessarily have to be commented, but `fatal` and `error` yes